tests/python/rest_api/test_users.py

```diff
@@ -26,41 +38,37 @@ def _test_can_see(
         **kwargs,
     ):
         with make_api_client(user) as api_client:
-            # TODO: refactor into several functions
             if id_ == "self":
-                (_, response) = api_client.users_api.retrieve_self(**kwargs, _parse_response=False)
+                response = self._case_self_id(api_client, **kwargs)
                 assert response.status == HTTPStatus.OK
                 response_data = json.loads(response.data)
             elif id_ is None:
-                response_data = get_paginated_collection(
-                    api_client.users_api.list_endpoint, return_json=True, **kwargs
-                )
+                response_data = self._case_none_id(api_client, **kwargs)
             else:
-                (_, response) = api_client.users_api.retrieve(id_, **kwargs, _parse_response=False)
+                response = self._case_other_id(api_client, id_, **kwargs)
                 assert response.status == HTTPStatus.OK
                 response_data = json.loads(response.data)
-
         assert DeepDiff(data, response_data, ignore_order=True, exclude_paths=exclude_paths) == {}
 
     def _test_cannot_see(
         self, user, id_: typing.Union[typing.Literal["self"], int, None] = None, **kwargs
     ):
         with make_api_client(user) as api_client:
-            # TODO: refactor into several functions
             if id_ == "self":
-                (_, response) = api_client.users_api.retrieve_self(
-                    **kwargs, _parse_response=False, _check_status=False
-                )
+                return self._case_self_id(api_client, **kwargs, _check_status=False)
             elif id_ is None:
                 (_, response) = api_client.users_api.list(
                     **kwargs, _parse_response=False, _check_status=False
                 )
             else:
-                (_, response) = api_client.users_api.retrieve(
-                    id_, **kwargs, _parse_response=False, _check_status=False
-                )
+                return self._case_other_id(api_client, id_, **kwargs, _check_status=False)
             assert response.status == HTTPStatus.FORBIDDEN
 
+    def _test_get_user_info(self, user):
```

Comment:
What is the difference between the test and `def test_everybody_can_see_self(self, users_by_name)`? Could you please explain?

Reply:
This test also verifies the number of response parameters and the requested username

Reply:
@awbage , do you think the number of response parameters is really important? I hope on Wednesday to hear more on the topic.